### DISCUSSION TITLE
Where should I store recipe steps?

### DISCUSSION THREAD

**Discussion by nikbakhts** [OrchardCMS/OrchardCore#6938] [Q&A]
To use some of Orchard Core Modules the documents shows what recipe steps must be added. What I couldn't figure out from the documents is where these steps are supposed to be stored. For instance in OpenId module, the document gives a sample for OpenID Connect Client Settings recipe step but I don't know where to store it. 

Do I need to save them in a json file in the root folder of my application? What should I name the file?

**Comment by deanmarcussen** [comment]
It depends if you are using recipe migrations, or just recipes during setup.

Recipes (for setup, or use through the admin later) can be stored in your `Theme` or `Module` under the `Recipes` folder, and by convention they are named `...recipe.json`

Recipe Migrations are stored in a `Migrations` folder and also named `...recipe.json`

**Reply by nikbakhts** [reply]
Thanks for your response Dean. I assume those folders are available in CMS.

I'm trying to use the framework only, as explained in this document https://docs.orchardcore.net/en/dev/docs/guides/create-modular-application-mvc/. 

I was successful in creating custom modules and using them from my modular application but not for OrchardCore modules. I assume the first step is the same, adding the module package to my modular project. But then I'm not sure for a module like OpenId where I can set its configuration values.
